Firestore/core/src/remote/serializer.cc

```diff
@@ -1244,7 +1207,7 @@ FilterList Serializer::DecodeCompositeFilter(
   FilterList result;
   result = result.reserve(composite.filters_count);
 
-  for (pb_size_t i = 0; reader->ok() && i != composite.filters_count; ++i) {
+  for (pb_size_t i = 0; i != composite.filters_count; ++i) {
```

Comment:
Question: what is the reason for this change?

Reply:
It simplifies the loop. The check should no longer be needed since an `InvalidFieldPath` is semantically valid in the consuming types (FieldMask, QueryTarget). I can change this back if exiting early is a better strategy here.

Reply:
I presume exiting early is better from the performance perspective, though now that I think about it, it only matters in the error case which should be rare. I'll leave it up to you.